components/renku_data_services/secrets/models.py

```diff
@@ -15,6 +17,7 @@ class Secret(BaseModel):
     encrypted_key: bytes = Field(repr=False)
     id: str | None = Field(default=None, init=False)
     modification_date: datetime = Field(default_factory=lambda: datetime.now(UTC).replace(microsecond=0), init=False)
+    kind: SecretKind
```

Comment:
suggestion(blocking): The base model should not use/depend on auto-generated code from the apispec. We should define an identical enum here and use it in the model. Also we should use this model enum in all other components like the blueprints. Let the autogenerated enum just be for serialization/deserialization. The problem with the autogenerated enum is that when it changes it will be automatically regenerated so you may get surprised with new fields or more drastic changes you dont expect. Also since the apispec stuff and migrations are autogenerated it can happen that you autogenerate a new enum from an apispec and also autogenerate a migration that is not fully expected/intended.

I know we do this in a few places already. I plan to go in and change all occurrences like what I described above during cooldown.

Reply:
I've added an enum to models. However, we cannot pass the `Secret` model to apispec `SecretWithId` model's `model_validate` anymore when returning results from various endpoints (since now they use different `SecretKind` enums). To address this, I call `Secret.model_dump_json` on the secret model. I then don't call the `apispec.SecretWithId.model_validate` on the apispec model since imo we already have a valid json result and don't need to validate it (and dump it to json) again. I can put back this in, if you think it's required.